lib/ansible/module_utils/cloudstack.py

```diff
@@ -143,8 +143,8 @@ def _connect(self):
             'endpoint': self.module.params.get('api_url') or config.get('endpoint'),
             'key': self.module.params.get('api_key') or config.get('key'),
             'secret': self.module.params.get('api_secret') or config.get('secret'),
-            'timeout': self.module.params.get('api_timeout') or config.get('timeout'),
-            'method': self.module.params.get('api_http_method') or config.get('method'),
+            'timeout': self.module.params.get('api_timeout') or config.get('timeout') or 10,
```

Comment:
In order to allow zero value for `timeout` key, should not that be like this ?
```
'timeout': self.module.params.get('api_timeout', config.get('timeout', 10)),
```

Reply:
I see your point, but timeout 0 looks not something useful to me for a real live use case. 
Could be probably helpful for integration tests in terms of "what happens if the request times out".

Reply:
Indeed. Besides current code should not be updated: zero isn't a valid ([1](https://github.com/exoscale/cs/blob/master/cs/client.py#L136)) value ([2](https://github.com/requests/requests/blob/master/requests/adapters.py#L431)):

```
In [3]: from urllib3.util import Timeout
In [4]: timeout = Timeout(connect=0, read=0)
[...]
ValueError: Attempted to set connect timeout to 0, but the timeout cannot be set to a value less than or equal to 0.
```

Reply:
👌